hdk/jack_analyzer/syntax.py

```diff
@@ -0,0 +1,340 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import TypeAlias
+from xml.dom.minidom import Document, Element
+from enum import Enum
+
+
+def _add_child(element: Element, tag_name: str, value: str | None) -> Element:
+    dom_tree = element.ownerDocument
+    child = dom_tree.createElement(tag_name)
+    child.appendChild(dom_tree.createTextNode(value))
+    element.appendChild(child)
+    return child
+
+
+@dataclass(frozen=True)
+class ParameterList:
+    parameters: list[tuple[str, str]]
+
+    def export_to_xml(self, dom_tree: Document) -> Element:
+        element = dom_tree.createElement("parameterList")
+        for i, (_type, var_name) in enumerate(self.parameters):
+            element_type = (
+                "keyword" if _type in {"int", "char", "boolean"}
+                else "identifier"
+            )
+            _add_child(element, element_type, _type)
+            _add_child(element, "identifier", var_name)
+            if i < len(self.parameters) - 1:
+                _add_child(element, "symbol", ",")
+        return element
+
+
+class SimpleTermType(Enum):
+    KEYWORD_CONSTANT = 0
+    INTEGER_CONSTANT = 1
+    STRING_CONSTANT = 2
+    VAR_NAME = 3
+
+
+@dataclass(frozen=True)
+class SimpleTerm:
```

Comment:
Split it to ConstantTerm and VarTerm. War term should contain attributes name: str and indexer: Expression | None.